tests/test_messages_queue.cc

```diff
@@ -0,0 +1,482 @@
+  ScopedRemoveFile remover(file);
+  TEST_EQUAL(true, FileManager::AppendStringToFile(file, file));
+  TEST_EQUAL(file, FileManager::ReadFileAsString(file));
+}
+
+void Test_ForEachFileInDir() {
+  {
+    bool was_called_at_least_once = false;
+    FileManager::ForEachFileInDir("", [&was_called_at_least_once](const std::string &) -> bool {
+      was_called_at_least_once = true;
+      return true;
+    });
+    TEST_EQUAL(false, was_called_at_least_once);
+  }
+
+  {
+    std::vector<std::string> files, files_copy;
+    std::vector<std::unique_ptr<ScopedRemoveFile>> removers;
+    for (size_t i = 0; i < 5; ++i) {
+      const std::string file = GenerateTemporaryFileName();
+      files.push_back(file);
+      removers.emplace_back(new ScopedRemoveFile(file));
+      TEST_EQUAL(true, FileManager::AppendStringToFile(file, file));
+    }
+    files_copy = files;
+    const std::string directory = FileManager::GetDirectoryFromFilePath(files[0]);
+    TEST_EQUAL(false, directory.empty());
+    FileManager::ForEachFileInDir(directory, [&files_copy](const std::string & path) -> bool {
+      // Some random files can remain in the temporary directory.
+      const auto found = std::find(files_copy.begin(), files_copy.end(), path);
+      if (found != files_copy.end()) {
+        TEST_EQUAL(path, FileManager::ReadFileAsString(path));
+        files_copy.erase(found);
+      }
+      return true;
+    });
+    TEST_EQUAL(size_t(0), files_copy.size());
+
+    // Test if ForEachFileInDir can be correctly interrupted in the middle.
+    files_copy = files;
+    FileManager::ForEachFileInDir(directory, [&files_copy](const std::string & path) -> bool {
+      // Some random files can remain in the temporary directory.
+      const auto found = std::find(files_copy.begin(), files_copy.end(), path);
+      if (found != files_copy.end()) {
+        std::remove(path.c_str());
+        files_copy.erase(found);
+        if (files_copy.size() == 1) {
+          return false;  // Interrupt when only 1 file left
+        }
+      }
+      return true;
+    });
+    TEST_EQUAL(size_t(1), files_copy.size());
+    // At this point, only 1 file should left in the folder.
+    for (const auto & file : files) {
+      if (file == files_copy.front()) {
+        TEST_EQUAL(file, FileManager::ReadFileAsString(file));
+      } else {
+        TEST_EQUAL("", FileManager::ReadFileAsString(file))
+      }
+    }
+  }
+}
+
+void Test_GetFileSize() {
+  const std::string file = GenerateTemporaryFileName();
+  ScopedRemoveFile remover(file);
+  // File does not exist yet.
+  TEST_EQUAL(-1, FileManager::GetFileSize(file));
+  // Use file name itself as a file contents.
+  TEST_EQUAL(true, FileManager::AppendStringToFile(file, file));
+  TEST_EQUAL(static_cast<int64_t>(file.size()), FileManager::GetFileSize(file));
+  // It should fail for directories.
+  TEST_EQUAL(-1, FileManager::GetFileSize(FileManager::GetDirectoryFromFilePath(file)));
```

Comment:
After checking the code I have found that it is also used in ReadFileAsString implementation, so probably will consider it later.

> On Jun 10, 2015, at 20:14, Dima <notifications@github.com> wrote:
> 
> In tests/test_messages_queue.cc <https://github.com/KnowSheet/Alohalytics/pull/36#discussion_r32141530>:
> 
> > +        TEST_EQUAL("", FileManager::ReadFileAsString(file))
> > +      }
> > +    }
> > +  }
> > +}
> > +
> > +void Test_GetFileSize() {
> > +  const std::string file = GenerateTemporaryFileName();
> > +  ScopedRemoveFile remover(file);
> > +  // File does not exist yet.
> > +  TEST_EQUAL(-1, FileManager::GetFileSize(file));
> > +  // Use file name itself as a file contents.
> > +  TEST_EQUAL(true, FileManager::AppendStringToFile(file, file));
> > +  TEST_EQUAL(static_cast<int64_t>(file.size()), FileManager::GetFileSize(file));
> > +  // It should fail for directories.
> > +  TEST_EQUAL(-1, FileManager::GetFileSize(FileManager::GetDirectoryFromFilePath(file)));
> Context: This is the only time you compare GetFileSize() against negative one. :-)
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/KnowSheet/Alohalytics/pull/36/files#r32141530>.
>